plugins/out_es/es_conf.c

```diff
@@ -24,10 +24,101 @@
 #include <fluent-bit/flb_http_client.h>
 #include <fluent-bit/flb_signv4.h>
 #include <fluent-bit/flb_aws_credentials.h>
+#include <mbedtls/base64.h>
 
 #include "es.h"
 #include "es_conf.h"
 
+/*
+ extract_cloud_host extracts the public hostname
+ of a deployment from a Cloud ID string.
+
+ The Cloud ID string has the format "<deployment_name>:<base64_info>".
+ Once decoded, the "base64_info" string has the format "<deployment_region>$<elasticsearch_hostname>$<kibana_hostname>"
+ and the function returns "<elasticsearch_hostname>.<deployment_region>" token.
+*/
+int extract_cloud_host(struct flb_elasticsearch *ctx, char *cloud_id) {
+    ssize_t ret;
+    size_t olen = 180;
+    int items = 0;
+    char *host_hash = NULL;
+    char *host_region = NULL;
+    unsigned char dec_cloud_id[180] = { 0 };
+    struct mk_list *cloud_id_toks = NULL;
+    struct mk_list *cloud_host_toks = NULL;
+    struct mk_list *head = NULL;
+    struct flb_split_entry *entry = NULL;
+    struct mk_list *host_head = NULL;
+
+    /* keep only part after first ":" */
+    cloud_id_toks = flb_utils_split(cloud_id, ':', 1);
```

Comment:
Clever approach, thanks! I switch to it, with some modifications since `hash` and `region` should be inverted to shape the final host string. I used `strtok` to split and get only the first part and add in the end of the host-hash string.